### DISCUSSION TITLE
Repeater disconnected on self hosted instance

### DISCUSSION THREAD

**Discussion by f4iey** [Vail-CW/vail_repeater_depricated#64] [Q&A]
Hello, 
I just tried to deploy the Docker image on my homeserver: [cw.f4iey.fr](https://cw.f4iey.fr)

But for some reason, when I try to go in any channel (except *echo*), it keeps displaying *Repeater disconnected*.

Is there some configuration to do so I can get rid of the problem? I have no idea what is causing this.

If someone could give me some clues to fix it, that will be very appreciated!
73 TU DE F4IEY Jules.

**Comment by nealey** [comment]
I'm looking into this: my best guess right now is that it's something to do with CloudFlare front-ending your box and not passing through WebSockets. I'm moving this to a discussion for now, if it actually turns out to be a bug, I'll move it back to issues :)

**Reply by nealey** [reply]
@f4iey My Chrome DevTools makes it look like it keeps trying, and failing, to connect to the repeater (server) with WebSockets (wss://cw.f4iey.fr/chat?repeater=General).

Vail consists of two parts:

1. The client, which is a bunch of static HTML, CSS, and JavaScript
2. The server, which provides the static content over HTTP, and handles "Vail connections" over WebSockets.

What you're seeing is that part 1 is working: it's serving up the client content just fine. Part 2 is failing: the WebSocket connection keeps failing, and it keeps retrying. So we need to zero in on why the WebSocket connection would fail, when the HTTPS connection works.

WebSockets actually happen at a layer above HTTPS. The browser connects over HTTPS, and then says "I would like a WebSocket, please". Then the web server can either say "okay, now we're doing WebSockets", return a failure, or just drop the connection. I *think* it's dropping the connection, but I'm not entirely sure, since I'm not super skilled at debugging WebSockets. I have asked some friends to help me out here, I'll let you know what they say if anyone ever answers the request :)

The IP addresses for cw.f4iey.fr all belong to CloudFlare. Is there a way you can tell CloudFlare to pass through WebSockets?

**Reply by f4iey** [reply]
Thanks a lot for the answer :)

From what I looked up, Cloudflare indeed supports websockets by default.

You may be right, I surely need to do something with my Cloudflare Configuration.
When trying out locally, it worked no problem.

 I'll also try to figure something out.

73
